client/layout/masterbar/style.scss

```
@@ -9,10 +9,14 @@ $masterbar-color-secondary: #101517;
 @import "calypso/assets/stylesheets/shared/animation";
 
 // Hide the masterbar on WP Mobile App views.
-.is-mobile-app-view {
+body.is-mobile-app-view {
 	.masterbar {
 		display: none;
 	}
+	/* stylelint-disable-next-line length-zero-no-unit */
+	--masterbar-checkout-height: 0px;
+	/* stylelint-disable-next-line length-zero-no-unit */
+	--masterbar-height: 0px;
```

There's no need to disable the `length-zero-no-unit` rule.  In CSS, `0px` is the same as `0` and the `length-zero-no-unit` rule was trying to suggest that you should just go with `0` in that case.

```suggestion
	--masterbar-checkout-height: 0;
	--masterbar-height: 0;
```

---

Actually, scratch that. I see the usage in `calc()` and verify that it needs `px` to work well inside a calc.

---

Maybe we can add a comment here just like in #81785 to clarify why we need the unit.

---

This should be resolved now. Thanks for the review @tyxla!